config/webhook/manifests.yaml

```diff
@@ -295,6 +339,26 @@ webhooks:
     resources:
     - azuremanagedclusters
   sideEffects: None
+- admissionReviewVersions:
+  - v1
+  - v1beta1
+  clientConfig:
+    service:
+      name: webhook-service
+      namespace: system
+      path: /validate-infrastructure-cluster-x-k8s-io-v1beta1-azuremanagedclustertemplate
+  failurePolicy: Fail
```

Comment:
According to the [kubebuilder book](https://book.kubebuilder.io/reference/markers/webhook.html), the default value of `matchPolicy` is `exact` meaning it will match only if it exactly matches the specified rule. `Equivalent` matchrd a request if it modifies a resource listed in rules, even via another API group or version. I think this means it's only necessary for types that exist in previous API versions? So I think it should be good to remove